### ISSUE TITLE
[a11y][Violation][4.1.2 Name, Role, Value] Issues reported on Notebook UI using IBM Equal Access Accessibility Checker tool

### ISSUE THREAD

**Issue by nkn2022** [jupyter/notebook#7488]
## Description

I am running an accessibility test suite with Notebook UI using the **IBM Equal Access Accessibility Checker** tool on Chrome browser. The results look good overall, only a few things are missing to reach a first formal level of compliance. So kindly help look into these. 

Issues reported were violating **WCAG 2.2** requirements. 
Ref: https://www.ibm.com/able/requirements/checker-rule-sets

Full report (includes information on element locations where issue were found): 
[Accessibility_Report-7a73e10e-f8a4-4a06-a78d-5174a1309c5b.xlsx](https://github.com/user-attachments/files/17400350/Accessibility_Report-7a73e10e-f8a4-4a06-a78d-5174a1309c5b.xlsx)

# Category: `4.1.2 Name, Role, Value`
Ref: https://www.ibm.com/able/requirements/requirements/?version=v7_3#4_1_2

****Violation of Rule:** `element_tabbable_role_valid`**

> List of issues with elements that violate this rule with a screenshots
> > **Issue 1** 
> > The tabbable element's role 'generic' is not a widget role
> > **Element location:** 
> > `<div data-windowed-list-index="0" tabindex="0" aria-label="Code Cell Content with Output" class="lm-Widget jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-selected jp-mod-active">`
> > ![Image](https://github.com/user-attachments/assets/d54c2022-d19e-44a9-9414-c13e1049bb2c)

> > **Issue 2** 
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<jp-button minimal="" appearance="stealth" current-value="" tabindex="0" title="Create a duplicate of this cell below" data-command="notebook:duplicate-below" aria-label="Create a duplicate of this cell below" aria-disabled="false" class="jp-ToolbarButtonComponent">`
> > **Screenshot:**
> > ![Image](https://github.com/user-attachments/assets/8981ca18-2369-4977-a13d-b00bfc35d79a)

 ****Violation of Rule: `input_label_exists`**
> List of issues with elements that violate this rule with a screenshots
> > **Issue 1** 
> > Form control with "textbox" role has no associated label
> > **Element location:** 
> > `<div data-language="python" style="tab-size: 4;" tabindex="-1" aria-multiline="true" role="textbox" class="cm-content" contenteditable="true" translate="no" autocapitalize="off" autocorrect="off" spellcheck="false">`
> > **Screenshot:**
> > ![Image](https://github.com/user-attachments/assets/ee848a39-fdeb-4788-8889-daa44a54af78)

## Reproduce

- Install Chrome extension for IBM accessibility checker tool. Ref: https://www.ibm.com/able/toolkit/verify/automated
- Open Notebook UI on Chrome browser. Add some cells with code.
- Run the tool and we can see the above `Violation`s reported.

Here is the screenshot of the UI scanned for the attached report:
![Image](https://github.com/user-attachments/assets/895eebec-51ad-43cd-93bb-99c4a8193b68)

## Expected behavior

No violations are reported on the scanning to reach a first formal level of compliance.

**Comment by nkn2022** [comment]
Reference on what to do to fix this issue can be found here: 
https://www.ibm.com/able/toolkit/develop/user-input/#standard-component
https://www.ibm.com/able/toolkit/develop/user-input/#custom-components

**Comment by nkn2022** [comment]
**Update on this issue:** 
Tested with latest version 7.4.4 and these mentioned issues still exists.

**Comment by nkn2022** [comment]
Tested on latest version 7.4.7. Still see these issues. The parent issue has been updated: https://github.com/jupyterlab/jupyterlab/issues/16890#issuecomment-3519050918
Will keep this issue open.
